app/studio/[studioID]/addclass/page.tsx

```diff
@@ -0,0 +1,190 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import React, { useState, ChangeEvent, FormEvent } from 'react';
+import supabase from '../../../../lib/supabase';
+
+// To-Do:
+// Refactor location to pull from existing studio locations
+// Refactor tags section to be more user-friendly
+// Handle UTC timezome consistency for class start time
+// Route based on studio id, not hard-coded studio id
+
+type FormData = {
+  location: string;
+  class_name: string;
+  class_description: string;
+  class_date: string;
+  class_start: string;
+  class_duration: number;
+  class_tags: string;
+  instructor: string;
+}
+
+export default function AddClass() {
+  const router = useRouter();
+
+  const [formData, setFormData] = useState<FormData>({
+    location: '1',
+    class_name: '',
+    class_description: '',
+    class_date: '',
+    class_start: '',
+    class_duration: 0,
+    class_tags: '',
+    instructor: '',
+  });
+
+  // To be refactored to fetch studio locations on form load
+  // const fetchStudioLocations = async () => {
+  //   const { data, error } = await supabase
+  //     .from('locations')
+  //     .select('location_id, name')
+  //     .eq('studio_id', '1');
+  //   if (error) {
+  //     // console.error(error);
+  //   } else {
+  //     // console.log('Fetch Data: ', data);
+  //   }
+  // };
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
change event type to handle Input element and Text Area Element to fix type error
e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>

Reply:
Thank you for catching this! Change made!